packages/app/components/token-picker/TokenPicker.tsx

```diff
@@ -92,17 +160,58 @@ const TokenListRow = ({ closeAction, token }: TokenListRowProps) => (
   </div>
 );
 
-const TokenList = ({ closeAction, tokenList }: TokenListProps) => (
-  <div
-    className="mt-5 overflow-y-auto h-72 border-t border-surface-50 divide-y divide-surface-50"
-    id="tokenPickerList"
-  >
-    {tokenList.map(token => (
-      <TokenListRow
-        closeAction={closeAction}
-        key={token.address}
-        token={token}
-      />
-    ))}
-  </div>
-);
+const TokenList = ({
+  closeAction,
+  onClearSearchClick,
+  tokenList,
+  tokenSearchQuery
+}: TokenListProps) => {
+  const [filteredTokenList, setFilteredTokenList] = useState<Token[]>(
+    tokenList
+  );
+
+  const handleClearSearchClick = () => {
```

Comment:
I would remove the `click` on the name, I don't think it's relevant.

I can see it's a click on the button bellow if I want to. But It's not important to reason about this code.

```suggestion
  const handleClearSearch = () => {
```